feat: add plural support

jkalberer opened the pull request:
### Internal Release Notes
This PR does a few things

1. Support `count` and `ordinal` for i18n `t` function. _This does not add support on `Trans` components because there wasn't a good example on what properties to set to support defaults_
2. Remove the opinionated logic around `***_one` derived keys. These were being set as "not derived" and given specific logic to populate them with default values. This doesn't make a lot of sense since we generally will want to add values to all derived keys or none of them.
3. Derived keys no longer use `key` instead of `cleanKey` when setting the default value. This means that you'll end up with keys/value mapping
```
"key_one" => "key"
"key_other" => "key"
```
instead of 
```
"key_one" => "key_one"
"key_other" => "key_other"
```
I can't think of any case where I'd want to map the key to value like this. It's better to use the `key` as the fallback because if you have `keyAsDefaultValue` set, you want the keys to actually work as translations.

### Validation Steps
`yarn run test`

jkalberer committed:
PORTAL-6292 | @jkalberer | Add plural support to babel i18n extractor

jkalberer committed:
Apply suggestions from code review

jkalberer committed:
Merge pull request #1 from audioeye/plural_support

 PORTAL-6292 | @jkalberer | Add plural support to babel i18n extractor

jkalberer commented:
This resolves https://github.com/gilbsgilbs/babel-plugin-i18next-extract/issues/207 and https://github.com/gilbsgilbs/babel-plugin-i18next-extract/issues/167

gilbsgilbs reviewed (changes requested):
Thank you for this work. The PR is not too large so that's fine, but I'd appreciate more atomic PRs in the future, because my brain is no longer "hot" on that project. Anyways.

> Remove the opinionated logic around ***_one derived keys. These were being set as "not derived" and given specific logic to populate them with default values. This doesn't make a lot of sense since we generally will want to add values to all derived keys or none of them.

I don't want to bikeshed this and I don't have strong feeling against it, but as it is sort of a breaking change (arguably one I'd be fairly confident making without a major version bump), I'd prefer stronger incentives. I can't say I'm very convinced by the "it clearly doesn't make sense because _generally_ \<some projection on what people are doing\>" take…

Why did you implement it? Did it make code easier to write? Do you consider the old behavior a bug? I don't really see what value this change brings.

jkalberer committed:
PORTAL-6292 | @jkalberer | Support plurals with keys and default values

jkalberer committed:
PORTAL-6292 | @jkalberer | Refresh values when useI18nextDefaultValue is true

jkalberer committed:
PORTAL-6292 | @jkalberer | Add extraction for class members with the i18n object

dependabot[bot] committed:
chore(deps): bump brace-expansion

Bumps the npm_and_yarn group with 1 update in the /docs directory: [brace-expansion](https://github.com/juliangruber/brace-expansion).


Updates `brace-expansion` from 1.1.11 to 1.1.12
- [Release notes](https://github.com/juliangruber/brace-expansion/releases)
- [Commits](https://github.com/juliangruber/brace-expansion/compare/1.1.11...v1.1.12)

---
updated-dependencies:
- dependency-name: brace-expansion
  dependency-version: 1.1.12
  dependency-type: indirect
  dependency-group: npm_and_yarn
...

Signed-off-by: dependabot[bot] <support@github.com>

dmoxon committed:
IN-15402 | @dmoxon | @eslint/plugin-kit vulnerability (#6)

### TL;DR

Update ESLint and related dependencies to the latest versions.

### What changed?

- Updated `@eslint/js` from v9.16.0 to v9.34.0
- Updated `eslint` from v9.16.0 to v9.34.0
- Updated related ESLint dependencies:
  - `@eslint/config-array` to v0.21.0
  - `@eslint/config-helpers` to v0.3.1
  - `@eslint/core` to v0.15.2
  - `@eslint/eslintrc` to v3.3.1
  - `@eslint/object-schema` to v2.1.6
  - `@eslint/plugin-kit` to v0.3.5
  - `@humanwhocodes/retry` to v0.4.3
  - `eslint-scope` to v8.4.0
  - `eslint-visitor-keys` to v4.2.1
  - `espree` to v10.4.0

### How to test?

1. Run `yarn install` to update the dependencies
2. Run linting with `yarn lint` to ensure everything works correctly
3. Verify that no new linting errors are introduced

### Why make this change?

Keeping ESLint and its dependencies up to date ensures we have the latest bug fixes, performance improvements, and security patches. This update also maintains compatibility with our current codebase while providing access to the latest features and improvements in the ESLint ecosystem.

dependabot[bot] committed:
chore(deps): bump the npm_and_yarn group across 1 directory with 3 updates

Bumps the npm_and_yarn group with 3 updates in the / directory: [@babel/helpers](https://github.com/babel/babel/tree/HEAD/packages/babel-helpers), [@babel/runtime](https://github.com/babel/babel/tree/HEAD/packages/babel-runtime) and [brace-expansion](https://github.com/juliangruber/brace-expansion).


Updates `@babel/helpers` from 7.26.0 to 7.28.3
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.28.3/packages/babel-helpers)

Updates `@babel/runtime` from 7.26.0 to 7.28.3
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.28.3/packages/babel-runtime)

Updates `brace-expansion` from 1.1.11 to 1.1.12
- [Release notes](https://github.com/juliangruber/brace-expansion/releases)
- [Commits](https://github.com/juliangruber/brace-expansion/compare/1.1.11...v1.1.12)

---
updated-dependencies:
- dependency-name: "@babel/helpers"
  dependency-version: 7.28.3
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: "@babel/runtime"
  dependency-version: 7.28.3
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: brace-expansion
  dependency-version: 1.1.12
  dependency-type: indirect
  dependency-group: npm_and_yarn
...

Signed-off-by: dependabot[bot] <support@github.com>

dmoxon committed:
NOJIRA | @dmoxon | Create a publish github action (#8)

### TL;DR

Prepare package for publishing to GitHub Packages registry with AudioEye scoped name and add automated publishing workflow.

### What changed?

- Renamed package from `babel-plugin-i18next-extract` to `@audioeye/babel-plugin-i18next-extract`
- Updated version from `0.1.0` to `0.1.2`
- Added GitHub Packages registry configuration to `package.json`
- Created new GitHub workflow for automated publishing (`publish.yml`)
- Dropped Node.js 18 from the test matrix
- Updated import assertion syntax in rollup config from `assert` to `with`

### How to test?

1. Verify the package can be published to GitHub Packages by running the new workflow manually
2. Confirm the package can be installed from GitHub Packages registry
3. Ensure tests pass on Node.js versions 20, 22, and 23

### Why make this change?

This change prepares the package for distribution through GitHub Packages under the AudioEye organization scope. The automated publishing workflow will streamline version management and releases, allowing for controlled version bumps (patch, minor, major) through the GitHub Actions interface.

dmoxon committed:
NOJIRA | @dmoxon | Release bot auth for publish (#10)

### TL;DR

Enhanced the GitHub Actions publish workflow to support automated releases with proper authentication and versioning.

### What changed?

- Added trigger for the workflow on pushes to the master branch
- Added conditional to skip workflow when commit message contains "chore(lerna)"
- Implemented GitHub App token creation for authentication
- Configured Git user identity for the release bot
- Added registry URL configuration for GitHub packages
- Modified the versioning process to create Git tags
- Added commit and push steps to record version changes
- Updated the publish command to run from the lib directory

### How to test?

1. Merge this PR to master
2. Verify that the workflow runs automatically on push
3. Check that version bumps create proper Git tags
4. Confirm that packages are published correctly to the registry
5. Manually trigger the workflow with different version types (patch/minor/major)

### Why make this change?

This improves the release automation by properly authenticating with GitHub's protected branches, creating appropriate Git tags and commits for version changes, and ensuring the release process is properly tracked in the repository history. The conditional execution prevents infinite loops when the workflow itself commits changes.

gl3nn commented:
@jkalberer @gilbsgilbs Any update on this?

Could it perhaps be extracted so that it does the feature where `defaultValue_other` populates `key_other`? Right now when you use a `count` you get just `key_one` populated by the `defaultValue` and `key_other` is empty.

gilbsgilbs commented:
@gl3nn I'm not actively working on this. Feel free to rebase, address the comments, and submit a new PR. I'll happily move it forward.

jkalberer commented:
Yeah -- apologies. I haven't really checked on this since I opened it and we're just using our own branch internally.

If someone wants to take over this PR (splitting it up and addressing the feedback), go for it.

jkalberer commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "babel-plugin-i18next-extract",
-  "version": "0.1.0",
+  "name": "@audioeye/babel-plugin-i18next-extract",
```
Revert

jgonzalez-ae committed:
Adding overrides to fix vulnerabilities.

jgonzalez-ae committed:
Merge pull request #13 from audioeye/Vulnerability_fix_override

dependabot[bot] committed:
chore(deps): bump @isaacs/brace-expansion from 5.0.0 to 5.0.1 in /docs

Bumps @isaacs/brace-expansion from 5.0.0 to 5.0.1.

---
updated-dependencies:
- dependency-name: "@isaacs/brace-expansion"
  dependency-version: 5.0.1
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dfelgar-ae committed:
Merge pull request #15 from audioeye/dependabot/npm_and_yarn/docs/isaacs/brace-expansion-5.0.1

chore(deps): bump @isaacs/brace-expansion from 5.0.0 to 5.0.1 in /docs

dfelgar-ae committed:
Merge pull request #7 from audioeye/dependabot/npm_and_yarn/npm_and_yarn-ab8a323cdf

chore(deps): bump the npm_and_yarn group across 1 directory with 3 updates

jgonzalez-ae committed:
Updating vulnerabilities.

jgonzalez-ae committed:
Updated other vulnerabilities

jgonzalez-ae committed:
Updated necessary package.

jgonzalez-ae committed:
Merge pull request #18 from audioeye/Fixing_vulnerabilities_packages

IN-20738 | @jgonzalez-ae | Updating vulnerabilities.
